src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmVocabularyAnnotation.cs

```diff
@@ -44,8 +47,45 @@ public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term,
             this.term = term;
             this.qualifier = qualifier;
             this.value = value;
+            this.usesDefault = false;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the EdmVocabularyAnnotation class
+        /// to the default value.
+        /// </summary>
+        /// <param name=“target”>Element the annotation applies to.</param>
+        /// <param name=“term”>Term bound by the annotation</param>
+        public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term)
+        {
+            // Check arguments
+            EdmUtil.CheckArgumentNull(target, "target");
+            EdmUtil.CheckArgumentNull(term, "term");
+            if (term.DefaultValue == null)
+            {
+                throw new InvalidOperationException(Strings.EdmVocabularyAnnotations_DidNotFindDefaultValue(term.Name));
+            }
+
+            // Initialize
+            this.target = target;
+            this.term = term;
+            this.qualifier = null;
+            this.value = EdmTermExtensions.BuildDefaultEdmExpression(term.Type, term.DefaultValue);
+            this.usesDefault = true;

```

Comment:
this constructor copies most of the logic from the previous constructor. not a big deal, since the logic is neither complex nor long, but consider centralizing constructor logic, either by having one constructor call another with default values (as is done on line 29) or by having a common "initialize" method called by constructors.

Reply:
Okay, thanks! I had noticed that as well, but since I can't call another constructor in the middle of the constructor body, I hadn't done that -- but it makes sense to just create a helper method!